prov/sockets/src/sock_progress.c

```diff
@@ -2453,6 +2454,55 @@ static void sock_pe_poll(struct sock_pe *pe)
 	}
 }
 
+static void sock_pe_set_affinity(struct sock_pe *pe)
+{
+	char *saveptra, *saveptrb, *saveptrc;
+	char *a, *b, *c, *s;
+	int j, first, last,stride;
+	cpu_set_t mycpuset;
+	pthread_t mythread;
+
+	fi_var_register(&sock_prov, "pe_affinity",
+                        "Which core(s) to be used for binding progress thread\n" \
+			"\tUsage: core_id_start[-core_id_end[:stride]][,<next_rule>]");
+	if (fi_var_get_str(&sock_prov, "pe_affinity", &s) != FI_SUCCESS)
+		return;
+
+	mythread = pthread_self();
+	CPU_ZERO(&mycpuset);
+
+	a = strtok_r(s,",", &saveptra);
+	while (a)
+	{
+		first = last = -1;
+		stride = 1;
+		b = strtok_r(a,"-",&saveptrb);
+		first = atoi(b);
+		/* Check for range delimiter */
+		if ((b = strtok_r(NULL,"-",&saveptrb)))
+		{
+			c = strtok_r(b,":",&saveptrc);
+			last = atoi(c);
+			/* Check for stride */
+			if ((c = strtok_r(NULL,":",&saveptrc))) {
+				stride = atoi(c);
+			}
+		}
+		
+		if (last == -1)
+			last = first;
+		
+		for (j = first; j<= last; j+=stride)
+			CPU_SET(j, &mycpuset);
+		a =  strtok_r(NULL,",", &saveptra);
+	}
+	
+	j = pthread_setaffinity_np(mythread, sizeof(cpu_set_t), &mycpuset);
```

Comment:
@jsquyres - Thanks for catching it. I will protect this with a #ifdef

@shefty - This will not directly improve any performance, but just provides a means for setting affinity for the progress thread. Setting affinity for progress thread and the application thread will avoids variability in performance. Btw, I have verified that the affinity patch works fine (using pidstat).